libsoftwarecontainer/unit-test/filegatewayparser_unittest.cpp

```diff
@@ -139,6 +140,109 @@ TEST_F(FileGatewayParserTest, TestEmptyPathInContainer) {
         "}";
     convertToJSON(config);
 
-    ASSERT_EQ(ReturnCode::FAILURE, parser.parseFileGatewayConfigElement(configJSON, result));
+    ASSERT_EQ(ReturnCode::FAILURE, parser.parseConfigElement(configJSON, result));
 }
 
+/*
+ * Using the same configuration twice is ok, as long as the container path
+ * is the same
+ */
+TEST_F(FileGatewayParserTest, TestSameConfigTwice) {
+    const std::string config = 
+        "{"
+            "  \"path-host\": \"" + FILE_PATH + "\""
+            ", \"path-container\": \"" + CONTAINER_PATH + "\""
+        "}";
+    convertToJSON(config);
+
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.parseConfigElement(configJSON, result));
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.matchEntry(result, settings));
+    ASSERT_EQ(1u, settings.size());
+
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.parseConfigElement(configJSON, result));
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.matchEntry(result, settings));
+    ASSERT_EQ(1u, settings.size());
+}
+
+/*
+ * Using the same paths but with different values on read-only is fine, since we
+ * will use the most permissive setting.
+ */
+TEST_F(FileGatewayParserTest, TestSameConfigReadWrite) {
+    const std::string config1 =
+        "{"
+            "  \"path-host\": \"" + FILE_PATH + "\""
+            ", \"path-container\": \"" + CONTAINER_PATH + "\""
+            ", \"read-only\": true"
+        "}";
+    convertToJSON(config1);
+
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.parseConfigElement(configJSON, result));
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.matchEntry(result, settings));
+    ASSERT_EQ(1u, settings.size());
+
+    const std::string config2 =
+        "{"
+            "  \"path-host\": \"" + FILE_PATH + "\""
+            ", \"path-container\": \"" + CONTAINER_PATH + "\""
+            ", \"read-only\": false"
+        "}";
+    convertToJSON(config2);
+
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.parseConfigElement(configJSON, result));
+    ASSERT_EQ(ReturnCode::SUCCESS, parser.matchEntry(result, settings));
+    ASSERT_EQ(1u, settings.size());
+    ASSERT_EQ(settings.at(0).readOnly, false);
+}
+
+/*
+ * Testing same host path but different container paths is fine, we will
+ * mount the same file to two places, no problems.
+ */
+TEST_F(FileGatewayParserTest, TestSameHostPathDifferentContainerPath) {
```

Comment:
I think TestSame template is not good for those. ParameterHostPathOverload could be better fit imho.
And also since the test fixture is already mentioning this is a test maybe we can remove test word from test names. I mean wouldn't be better to filter like `FileGatewayParserTest.SameHostPathDifferentContainerPath` instead of `FileGatewayParserTest.TestSameHostPathDifferentContainerPath`